tests/test_pipeline_common.py

```diff
@@ -0,0 +1,235 @@
+import copy
+import importlib
+import os
+import random
+from pathlib import Path
+
+import requests
+
+from .pipelines.test_pipelines_audio_classification import AudioClassificationPipelineTests
+from .pipelines.test_pipelines_feature_extraction import FeatureExtractionPipelineTests
+from .pipelines.test_pipelines_fill_mask import FillMaskPipelineTests
+from .pipelines.test_pipelines_image_classification import ImageClassificationPipelineTests
+from .pipelines.test_pipelines_text_classification import TextClassificationPipelineTests
+from .pipelines.test_pipelines_text_generation import TextGenerationPipelineTests
+
+
+pipeline_test_mapping = {
+    "audio-classification": {"test": AudioClassificationPipelineTests},
+    "feature-extraction": {"test": FeatureExtractionPipelineTests},
+    "fill-mask": {"test": FillMaskPipelineTests},
+    "image-classification": {"test": ImageClassificationPipelineTests},
+    "text-classification": {"test": TextClassificationPipelineTests},
+    "text-generation": {"test": TextGenerationPipelineTests},
+}
+
+for task, task_info in pipeline_test_mapping.items():
+    test = task_info["test"]
+    task_info["mapping"] = {
+        "pt": getattr(test, "model_mapping", None),
+        "tf": getattr(test, "tf_model_mapping", None),
+    }
```

Comment:
lazy way to add these mappings from the (existing) explicit pipeline test classes